From 385957e7daa108c87e881d219d729e8c887e9089 Mon Sep 17 00:00:00 2001
From: Urvishkumar Pandya <quic_urvishku@quicinc.com>
Date: Tue, 3 Oct 2023 15:02:36 +0530
Subject: [PATCH] cfg80211: remove WARN_ON in cfg80211 connect result

found cfg80211_connect_result call trace  while configuring client AP.
disabled warning for cfg80211_connect_result and add print if bss is not found

call trace:
__cfg80211_connect_result+0x2e0/0x5ac
cfg80211_process_wdev_events+0xcc/0x164
cfg80211_process_rdev_events+0x38/0x44
cfg80211_event_work+0x24/0x38
process_one_work+0x1a0/0x228
worker_thread+0x1f4/0x264
kthread+0xe8/0xf8
ret_from_fork+0x10/0x20

Change-Id: I9e82abb9286725caeb0736e620a50d5848dbc83c
Signed-off-by: Urvishkumar Pandya <quic_urvishku@quicinc.com>
---
 net/wireless/sme.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/net/wireless/sme.c b/net/wireless/sme.c
index 6e87d2cd8345..fc71b90d1d20 100644
--- a/net/wireless/sme.c
+++ b/net/wireless/sme.c
@@ -836,8 +836,9 @@ void __cfg80211_connect_result(struct net_device *dev,
 		return;
 	}
 
-	if (WARN_ON(bss_not_found)) {
+	if (bss_not_found) {
 		cfg80211_connect_result_release_bsses(wdev, cr);
+		pr_warn("%s: bss not found\n", __func__);
 		return;
 	}
 
-- 
2.34.1

